apacheconfig/wloader.py

```diff
@@ -0,0 +1,204 @@
+            self._whitespace = raw[1]
+            self._raw = tuple(raw[2:])
+
+    @property
+    def ast_node_type(self):
+        """Returns object typestring as defined by the apacheconfig parser.
+
+        Returns:
+            The typestring (as defined by the apacheconfig parser) for
+            this node. The possible values for this are ``"comment"``,
+            ``"statement"``, ``"include"``, ``"includeoptional"``.
+        """
+        return self._type
+
+    @property
+    def whitespace(self):
+        """Returns preceding whitespace for this node.
+
+        For example::
+
+            ItemNode('\\n  option value').whitespace => "\\n  "
+            ItemNode('option value').whitespace => ""
+            ItemNode('\\n  # comment').whitespace => "\\n  "
+        """
+        return self._whitespace
+
+    @whitespace.setter
+    def whitespace(self, value):
+        """See base class. Operates on preceding whitespace."""
+        self._whitespace = value
+
+    @staticmethod
+    def parse(raw_str, options={}, parser=None):
+        """Factory for :class:`apacheconfig.ItemNode` by parsing data from a
+        config string.
+
+        Args:
+            options (dict): Additional options to pass to the created parser.
+                Ignored if another ``parser`` is supplied.
+            parser (:class:`apacheconfig.ApacheConfigParser`): optional, to
+                re-use an existing parser. If ``None``, creates a new one.
+        Returns:
+            :class:`apacheconfig.ItemNode` containing metadata parsed from
+            ``raw_str``.
+        """
+        if not parser:
+            parser = _create_apache_parser(options, start='startitem')
+        return ItemNode(parser.parse(raw_str))
+
+    @property
+    def name(self):
+        """Returns the name of this node.
+
+        The name is the first non-whitespace token in the directive. Cannot be
+        written. For comments, is the entire comment.
+        """
+        return self._raw[0]
+
+    @property
+    def has_value(self):
+        """Returns ``true`` if this :class:`apacheconfig.ItemNode` has a value.
+
+        ``ItemNode`` objects don't have to have a value, like option/value
+        directives with no value, or comments.
+        """
+        return len(self._raw) > 1
+
+    @property
+    def value(self):
+        """Returns the value of this item as a string.
+
+        The "value" is anything but the name. Can be overwritten.
+        """
+        if not self.has_value:
+            return None
+        return self._raw[-1]
+
+    @value.setter
+    def value(self, value):
+        """Sets for the value of this item.
+
+        Args:
+            value (str): string to set new value to.
+
+          .. todo:: (sydneyli) convert `value` to quotedstring when quoted
+        """
+        if not self.has_value:
+            self._raw = self._raw + (" ", value,)
+        self._raw = self._raw[0:-1] + (value,)
+
+    def dump(self):
```

Comment:
nit: would it make sense to have this as a property like `text` or something? Do you expect some parameters to appear in the future in this function?

Reply:
It might be good to leave it open to parameters in the future. Either way, I'm wary of making this an attribute in the interface since it actually walks through the entire entire tree to construct the dumped string.